src/python/pants/invalidation/build_invalidator.py

```diff
@@ -93,8 +94,30 @@ def key_for_target(self, target, transitive=False, fingerprint_strategy=None):
 class BuildInvalidator(object):
   """Invalidates build targets based on the SHA1 hash of source files and other inputs."""
 
-  def __init__(self, root):
-    self._root = os.path.join(root, GLOBAL_CACHE_KEY_GEN_VERSION)
+  class Factory(Subsystem):
+    options_scope = 'build-invalidator'
+
```

Comment:
It is a thing, console tasks mainly (all?) are examples. They don't fail at any rate, and since they already in-effect `--cache-ignore`, the behavior is a noop.

For example:
```
$ ./pants list --cache-ignore src/python/pants/{cache,invalidation}::
src/python/pants/invalidation:invalidation
src/python/pants/cache:cache
```